Pbi3 landmark list

skuuzie opened the pull request:
Proposed changes:
- Switch landmark detail page from fragment to activity
- Implement landmark detail repository and it's remote data source dependencies with dagger DI, in accordance to existing DI pattern/style
- Temporarily disable local database cache for landmark detail until further inspection
- Display landmark list on home fragment
- Landmark detail viewmodel is now functional to perform GET request from API

Minor changes:
- Reformat codes with built-in Android Studio reformatter
- Adjustment to UI (positioning, images, fonts, colors, dimens, styles, themes)
- Add a button on post-scan page to test out [Scan -> Landmark detail page] functionality

IbniShaquille committed:
Merge pull request #3 from EducaTour/staging

Staging: Sync to Main Branch

IbniShaquille committed:
Merge pull request #6 from EducaTour/staging

Staging

skuuzie committed:
remove unused file

skuuzie committed:
chore: add and update static files for ui (images, fonts, colors, styles, dimen)

skuuzie committed:
reformat code

skuuzie committed:
feat: display landmark list to home fragment

skuuzie committed:
chore: change fonts

skuuzie committed:
chore: landmark detail repository DI

skuuzie committed:
reformat code

skuuzie committed:
chore: landmark detail remote data source DI

skuuzie committed:
chore: provide landmark detail repository and viewmodel factory

skuuzie committed:
refactor: change landmark detail page to activity

skuuzie committed:
refactor: temporarily disable landmark detail local db cache

skuuzie committed:
fix: use retrofit's @Path instead of @Query

skuuzie committed:
fix: adjust data classes fields to match API's JSON response

skuuzie committed:
chore: add landmark detail activity to manifest

skuuzie committed:
chore: add test button to post-scan page

IbniShaquille commented on app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt:
```diff
@@ -4,34 +4,33 @@ import androidx.room.Database
         ContactInfo::class,
         Landmark::class,
-        LandmarkDetail::class,
+        // LandmarkDetail::class,
```
remove comment line

IbniShaquille commented on app/src/main/java/com/dicoding/motour/data/db/EducaTourDB.kt:
```diff
@@ -4,34 +4,33 @@ import androidx.room.Database
+abstract class EducaTourDB : RoomDatabase() {
     abstract fun landmarkDao(): LandmarkDao
-    abstract fun landmarkDetailDao(): LandmarkDetailDao
+
+    // abstract fun landmarkDetailDao(): LandmarkDetailDao
```
delete  comment line

IbniShaquille commented on app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt:
```diff
@@ -0,0 +1,34 @@
+        var landmark: LandmarkDetail? = null
+        try {
+            val response = landmarkDetailRemoteDatasource.getLandmarkDetail(id)
+            Log.d("LandmarkDetailRepositoryImpl", response.raw().toString())
```
for privacy issue, if this function is done, just remove the Logging

IbniShaquille commented on app/src/main/java/com/dicoding/motour/data/repository/landmark/detail/LandmarkDetailRepositoryImpl.kt:
```diff
@@ -0,0 +1,34 @@
+                landmark = body.landmark
+            }
+        } catch (e: Exception) {
+            Log.d("LandmarkDetailRepositoryImpl", e.message.toString())
```
remove this, or you can use Toast

IbniShaquille commented on app/src/main/java/com/dicoding/motour/presentation/di/core/DataBaseModule.kt:
```diff
@@ -20,10 +20,11 @@ class DataBaseModule {
     fun provideContactInfoDao(educaTourDB: EducaTourDB) = educaTourDB.contactInfoDao()
 
+    /*
     @Singleton
     @Provides
     fun provideLandmarkDetailDao(educaTourDB: EducaTourDB) = educaTourDB.landmarkDetailDao()
-
+    */
```
just remove

IbniShaquille commented on app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt:
```diff
@@ -32,15 +41,61 @@ class HomeFragment : Fragment() {
+                        }
+                    })
+
+                    Log.d("MYTAG", landmarkList.toString())
```
privacy issue, remove logging if done

IbniShaquille commented on app/src/main/java/com/dicoding/motour/presentation/home/HomeFragment.kt:
```diff
@@ -32,15 +41,61 @@ class HomeFragment : Fragment() {
+    }
+
+    private fun setLoading(state: Boolean) {
+        if (state) {
+            binding.progressIndicator.visibility = View.VISIBLE
+        } else {
+            binding.progressIndicator.visibility = View.GONE
+        }
```
for the simple mode use this one

binding.progressBar.visibility = if (isLoading) View.VISIBLE else View.GONE

IbniShaquille commented:
if class StringList is not used, just remove the class

IbniShaquille commented:
overalll, the code implementation is very good, goodjob!!!

skuuzie committed:
refactor: remove unused file and code as per code review

skuuzie committed:
chore: add & adjust fonts, drawables, strings

skuuzie committed:
refactor: add error message and retry mechanism for landmark list retrieval from API

skuuzie committed:
refactor: improve landmark list UI

skuuzie committed:
refactor: load image url from API to landmark list

skuuzie committed:
refactor: throw exception to display error on UI

skuuzie committed:
refactor: remove unused logging code

skuuzie commented:
Refactored the code as per code review. I've also improved the UI for landmark list fragment and added a custom error message for retry mechanism in case of network error

IbniShaquille commented:
okei, you can merge the PR
